### PR TITLE
Activity payload

### LINKED ISSUE #703: Payload for Activity
The `Activity` class currently doesn't have a payload property ([which they do in the api](https://developer.github.com/v3/activity/events/)), which would be really useful. I couldn't find any existing issues for this, is it something you guys have talked about in the past? 

I'd be happy to implement the payload for some of the most common [event types](https://developer.github.com/v3/activity/events/types/), but doing it for all types would be a big job (there are 25 types in total).

### PR DESCRIPTION
**Pull request by DaveWM**
resolves #703

### CODE REVIEW

**Comment by DaveWM** [comment]
There's a bit of a hack in the GitHubSerializerStrategy class (in SimpleJsonSerializer.cs) to get it to deserialize the payload to the correct type. Also, not sure if you'd prefer the tests for this to be in a separate file (currently in EventsClientsTests.cs).

File: Octokit/Models/Response/ActivityPayloads/ForkEventPayload.cs [hunk 1]
```diff
@@ -0,0 +1,12 @@
+﻿using System.Diagnostics;
+using System.Diagnostics.CodeAnalysis;
+
+namespace Octokit.Models.Response.ActivityPayloads
+{
+    [DebuggerDisplay("{DebuggerDisplay,nq}")]
+    public class ForkEventPayload : ActivityPayload
+    {
+        [SuppressMessage("Microsoft.Naming", "CA1704:IdentifiersShouldBeSpelledCorrectly", MessageId = "Forkee")]
```
Comment (shiftkey) [inline Octokit/Models/Response/ActivityPayloads/ForkEventPayload.cs]: You can add `Forkee` to the `CustomDictionary.xml` to make this error go away

File: Octokit.Tests/OctoKit.Tests-NetCore45.csproj [hunk 2]
```diff
@@ -167,6 +167,10 @@
       <Project>{c8bc13b6-3fa3-4716-827d-e7706f976fe1}</Project>
       <Name>Octokit-NetCore45</Name>
     </ProjectReference>
+    <ProjectReference Include="..\Octokit\Octokit.csproj">
```
Comment (shiftkey) [inline Octokit.Tests/OctoKit.Tests-NetCore45.csproj]: I'm not sure why these references are necessary.

Could you :fire: this one and the one in `Octokit.Tests-Portable`?

File: Octokit/Http/SimpleJsonSerializer.cs [hunk 3]
```diff
@@ -3,6 +3,8 @@
 using System.Diagnostics.CodeAnalysis;
 using System.Linq;
 using System.Reflection;
+using System.Text.RegularExpressions;
```
Comment (shiftkey) [inline Octokit/Http/SimpleJsonSerializer.cs]: :fire: unused reference

File: Octokit/Models/Response/ActivityPayloads/CommitCommentPayload.cs [hunk 4]
```diff
@@ -0,0 +1,10 @@
+﻿using System.Diagnostics;
+
+namespace Octokit.Models.Response.ActivityPayloads
```
Comment (shiftkey) [inline Octokit/Models/Response/ActivityPayloads/CommitCommentPayload.cs]: Can you drop all this excess namespace and just have `Octokit` here?

File: Octokit/Models/Response/ActivityPayloads/ForkEventPayload.cs [hunk 5]
```diff
@@ -0,0 +1,12 @@
+﻿using System.Diagnostics;
+using System.Diagnostics.CodeAnalysis;
+
+namespace Octokit.Models.Response.ActivityPayloads
```
Comment (shiftkey) [inline Octokit/Models/Response/ActivityPayloads/ForkEventPayload.cs]: :arrow_up:

File: Octokit/Models/Response/ActivityPayloads/IssueCommentPayload.cs [hunk 6]
```diff
@@ -0,0 +1,14 @@
+﻿using System.Diagnostics;
+
+namespace Octokit.Models.Response.ActivityPayloads
```
Comment (shiftkey) [inline Octokit/Models/Response/ActivityPayloads/IssueCommentPayload.cs]: :arrow_up:

File: Octokit/Models/Response/ActivityPayloads/IssueEventPayload.cs [hunk 7]
```diff
@@ -0,0 +1,13 @@
+﻿using System.Diagnostics;
+
+namespace Octokit.Models.Response.ActivityPayloads
```
Comment (shiftkey) [inline Octokit/Models/Response/ActivityPayloads/IssueEventPayload.cs]: :arrow_up:

File: Octokit/Models/Response/ActivityPayloads/PullRequestCommentPayload.cs [hunk 8]
```diff
@@ -0,0 +1,12 @@
+﻿using System.Diagnostics;
+
+namespace Octokit.Models.Response.ActivityPayloads
```
Comment (shiftkey) [inline Octokit/Models/Response/ActivityPayloads/PullRequestCommentPayload.cs]: :arrow_up:

File: Octokit/Models/Response/ActivityPayloads/PullRequestEventPayload.cs [hunk 9]
```diff
@@ -0,0 +1,13 @@
+﻿using System.Diagnostics;
+
+namespace Octokit.Models.Response.ActivityPayloads
```
Comment (shiftkey) [inline Octokit/Models/Response/ActivityPayloads/PullRequestEventPayload.cs]: :arrow_up:

File: Octokit/Models/Response/ActivityPayloads/PushEventPayload.cs [hunk 10]
```diff
@@ -0,0 +1,14 @@
+﻿using System.Collections.Generic;
+using System.Diagnostics;
+
+namespace Octokit.Models.Response.ActivityPayloads
```
Comment (shiftkey) [inline Octokit/Models/Response/ActivityPayloads/PushEventPayload.cs]: :arrow_up:

File: Octokit/Models/Response/ActivityPayloads/StarredEventPayload.cs [hunk 11]
```diff
@@ -0,0 +1,10 @@
+﻿using System.Diagnostics;
+
+namespace Octokit.Models.Response.ActivityPayloads
```
Comment (shiftkey) [inline Octokit/Models/Response/ActivityPayloads/StarredEventPayload.cs]: :arrow_up:

File: Octokit/Http/SimpleJsonSerializer.cs [hunk 12]
```diff
@@ -137,6 +155,30 @@ static string RemoveHyphenAndUnderscore(string stringValue)
                         p => p.JsonFieldName,
                         p => new KeyValuePair<Type, ReflectionUtils.SetDelegate>(p.Type, p.SetDelegate));
             }
+
+            private static Type GetPayloadType(string activityType)
```
Comment (shiftkey) [inline Octokit/Http/SimpleJsonSerializer.cs]: Would it be worth extracting this to an extension method for better readability?

**Comment by shiftkey** [comment]
> There's a bit of a hack in the GitHubSerializerStrategy class (in SimpleJsonSerializer.cs) to get it to deserialize the payload to the correct type.

That's the best place to put it. 

> Also, not sure if you'd prefer the tests for this to be in a separate file (currently in EventsClientsTests.cs).

Not really fussed by this. Would it be possible to get a couple of integration tests in place as well?

File: Octokit/Http/SimpleJsonSerializer.cs (hunk 12 above)
Comment (DaveWM) [inline Octokit/Http/SimpleJsonSerializer.cs]: Personally, I'd prefer to keep it how it is - but I'm happy to change it to an extension method if you want.

File: Octokit/Models/Response/ActivityPayloads/CommitCommentPayload.cs (hunk 4 above)
Comment (DaveWM) [inline Octokit/Models/Response/ActivityPayloads/CommitCommentPayload.cs]: Sure

File: Octokit.Tests/OctoKit.Tests-NetCore45.csproj (hunk 2 above)
Comment (DaveWM) [inline Octokit.Tests/OctoKit.Tests-NetCore45.csproj]: I'm going to blame resharper for this one :P

File: Octokit/Models/Response/ActivityPayloads/ForkEventPayload.cs (hunk 1 above)
Comment (DaveWM) [inline Octokit/Models/Response/ActivityPayloads/ForkEventPayload.cs]: Cool, will do

File: Octokit/Http/SimpleJsonSerializer.cs (hunk 12 above)
Comment (shiftkey) [inline Octokit/Http/SimpleJsonSerializer.cs]: Not something I'll block this on. I wonder if @haacked has any opinions on this.

File: Octokit.Tests/OctoKit.Tests-NetCore45.csproj (hunk 2 above)
Comment (shiftkey) [inline Octokit.Tests/OctoKit.Tests-NetCore45.csproj]: Suuuuuuuuure :P

**Comment by DaveWM** [comment]
I'll have a go at the integration tests for the simpler event types (commit comment, issue comment, etc.), but testing the events you get when you create a PR, push to a repo, fork a repo, etc. would be a lot harder.

**Comment by shiftkey** [comment]
> I'll have a go at the integration tests for the simpler event types (commit comment, issue comment, etc.), but testing the events you get when you create a PR, push to a repo, fork a repo, etc. would be a lot harder.

If you can find some archived events which you can fetch by a specific id that might be easier to test...

**Comment by DaveWM** [comment]
@shiftkey I've added the integration tests, they don't cover all the event types but they're better than nothing :)

**Comment by shiftkey** [comment]
Tests are :thumbsup: - just that change with `[Fact]` -> `[IntegrationTest]` to ensure credentials are setup right.

**Comment by DaveWM** [comment]
no problem - done :)

**Pull request merged**